### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by shinokada**
### 🔗 Related issue (optional)

Closes #

---

## 📑 Description

---

## 🔍 PR Type

- [ ] Bug fix
- [ ] Feature
- [ ] Documentation
- [ ] Refactor / Code cleanup
- [ ] Build / Tooling
- [ ] Other (please describe)

---

## 🚦 PR Status

- [ ] Draft (work in progress, not ready for review)
- [ ] Ready for review ✅

---

## ✅ Checklist

- [ ] My code follows the existing code style
- [ ] I have run `pnpm lint && pnpm check && pnpm test:e2e` and all tests pass
- [ ] CoderabbitAI review has been completed and **actionable suggestions were reviewed**
- [ ] I have updated documentation if my changes require it
- [ ] My PR is based on the latest `main` branch (not the published npm version)
- [ ] I have checked accessibility where applicable (ARIA, keyboard nav, etc.)
- [ ] I have reviewed the rendered component in the browser

---

## 🧪 Screenshots / Demos (optional)

---

## ⚠️ Breaking Changes (optional)

---

## Summary by CodeRabbit

* **New Features**
  * Expanded theming: multiple runtime palettes, new "load from static" theme option, richer color variants, and Popper prop to disable transitions.

* **Bug Fixes**
  * Improved focus/keyboard handling and more robust clipboard copy/write flows across overlays and dialogs.

* **Tests**
  * Large-scale additions: comprehensive test suites and helpers covering typography, media, layout, extend components, and utilities.

* **Style**
  * Typography metrics migrated to rem units; primary/brand color tokens refreshed.

* **Documentation**
  * Added guides on reactive contexts, test setup, timer usage, and a popover polyfill.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/lib/alert/theme.ts [hunk 1]
```diff
@@ -12,7 +12,26 @@ export const alert = tv({
       gray: "text-heading bg-neutral-secondary-medium",
       danger: "text-fg-danger-strong bg-danger-soft",
       success: "text-fg-success-strong bg-success-soft",
-      warning: "text-fg-warning bg-warning-soft"
+      warning: "text-fg-warning bg-warning-soft",
+            primary: "bg-primary-50 dark:bg-gray-800 text-primary-800 dark:text-primary-400",
+      secondary: "bg-secondary-50 dark:bg-secondary-800 text-secondary-800 dark:text-secondary-400",
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/alert/theme.ts]: fixed.

File: src/lib/alert/theme.ts [hunk 2]
```diff
@@ -108,6 +127,107 @@ export const alert = tv({
...
+    {
+      border: true,
+      color: "secondary",
+      class: "border-secondary-500 dark:border-secondary-200 divide-secondary-500 dark:divide-secondary-200"
+    },
+    {
+      border: true,
+      color: "gray",
+      class: "border-gray-300 dark:border-gray-800 divide-gray-300 dark:divide-gray-800"
+    },
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/alert/theme.ts:140]: fixed.

File: src/lib/badge/theme.ts [hunk 3]
```diff
@@ -94,6 +113,206 @@ export const badge = tv({
...
+    {
+      border: true,
+      color: "secondary",
+      class: "dark:bg-transparent dark:text-secondary-400 border-secondary-400 dark:border-secondary-400"
+    },
+    {
+      border: true,
+      color: "gray",
+      class: "dark:bg-transparent dark:text-gray-400 border-gray-400 dark:border-gray-400"
+    },
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/badge/theme.ts]: fixed.

File: src/tests/typography/span/span.test.ts [hunk 4]
```diff
@@ -0,0 +1,718 @@
...
+  });
+
+  describe("Edge Cases", () => {
+    test("renders correctly with empty children", () => {
+      const { container } = render(BasicSpanTest);
+      const span = container.querySelector('[data-scope="span"]');
+
+      expect(span).toBeInTheDocument();
+      expect(span?.tagName).toBe("SPAN");
+    });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/typography/span/span.test.ts]: Used option 1.

File: src/lib/badge/theme.ts [hunk 5]
```diff
@@ -94,6 +113,201 @@ export const badge = tv({
...
+    {
+      href: true,
+      color: "secondary",
+      class: "hover:bg-secondary-200 dark:hover:bg-secondary-700"
+    },
+    {
+      href: true,
+      color: "gray",
+      class: "hover:bg-gray-200 dark:hover:bg-gray-700"
+    },
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/badge/theme.ts]: fixed.

File: src/tests/extend/clipboard-manager/clipboard-manager.test.ts [hunk 6]
```diff
@@ -0,0 +1,588 @@
...
+      // Pin Item 2
+      const items = screen.getAllByRole("button", { name: /pin/i });
+      await user.click(items[1]); // Pin middle item
+
+      await waitFor(() => {
+        const allItems = screen.getAllByTestId("basic-clipboard");
+        const firstItemText = allItems[0].querySelector('[data-part="item-text"]')?.textContent;
+        expect(firstItemText).toBe("Item 2");
+      });
+    });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/clipboard-manager/clipboard-manager.test.ts:342]: fixed.

File: static/themes/default-runtime-original.css [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+  --leading-9: 36px; /* rem pls */
+  --leading-7: 28px;
+  --leading-8: 32px;
+  --leading-6: 24px;
+  --leading-4: 16px;
+  --leading-none: 1px;
+  --leading-5: 20px;
+  --tracking-tighter: -0.8px;
+  --leading-heading-none: 60px;
+  --tracking-tight: -0.4px;
```
[the remark this answers is not in the document]
Comment (shinokada) [reply static/themes/default-runtime-original.css]: fixed.

File: static/themes/default-runtime.css [hunk 8]
```diff
@@ -42,9 +71,9 @@
   --color-heading: var(--color-gray-900);
 
   /* used for custom brand colors */
-  --color-fg-brand-subtle: var(--color-blue-200);
-  --color-fg-brand: var(--color-blue-700);
-  --color-fg-brand-strong: var(--color-blue-900);
+  --color-fg-brand-subtle: var(--color-primary-200);
+  --color-fg-brand: var(--color-primary-700);
+  --color-fg-brand-strong: var(--color-primary-900);
```
[the remark this answers is not in the document]
Comment (shinokada) [reply static/themes/default-runtime.css:60]: fixed.

File: static/themes/default-runtime.css [hunk 9]
```diff
@@ -80,11 +109,11 @@
...
-  --color-brand-softer: var(--color-blue-50);
-  --color-brand-soft: var(--color-blue-100);
-  --color-brand: var(--color-blue-700);
-  --color-brand-medium: var(--color-blue-200);
-  --color-brand-strong: var(--color-blue-800);
+  --color-brand-softer: var(--color-primary-50);
+  --color-brand-soft: var(--color-primary-100);
+  --color-brand: var(--color-primary-700);
+  --color-brand-medium: var(--color-primary-200);
+  --color-brand-strong: var(--color-primary-800);
```
[the remark this answers is not in the document]
Comment (shinokada) [reply static/themes/default-runtime.css:100]: fixed.

File: static/themes/default-runtime.css [hunk 10]
```diff
@@ -177,11 +213,11 @@
...
-  --color-brand-softer: var(--color-blue-950);
-  --color-brand-soft: var(--color-blue-900);
-  --color-brand: var(--color-blue-600);
-  --color-brand-medium: var(--color-blue-900);
-  --color-brand-strong: var(--color-blue-700);
+  --color-brand-softer: var(--color-primary-950);
+  --color-brand-soft: var(--color-primary-900);
+  --color-brand: var(--color-primary-600);
+  --color-brand-medium: var(--color-primary-900);
+  --color-brand-strong: var(--color-primary-700);
```
[the remark this answers is not in the document]
Comment (shinokada) [reply static/themes/default-runtime.css:197]: fixed.

File: static/themes/default.css [hunk 11]
```diff
@@ -42,9 +71,9 @@
   --color-heading: var(--color-gray-900);
 
   /* used for custom brand colors */
-  --color-fg-brand-subtle: var(--color-blue-200);
-  --color-fg-brand: var(--color-blue-700);
-  --color-fg-brand-strong: var(--color-blue-900);
+  --color-fg-brand-subtle: var(--color-primary-200);
+  --color-fg-brand: var(--color-primary-700);
+  --color-fg-brand-strong: var(--color-primary-900);
```
[the remark this answers is not in the document]
Comment (shinokada) [reply static/themes/default.css:60]: fixed.

File: static/themes/enterprise-runtime.css [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+  --text-xs: 0.825rem;
+  --text-sm: 0.925rem;
+  --text-base: 1.125rem;
+
+  /* radius variables */
+  --radius-xxs: 1px;
+  --radius-xs: 1px;
+  --radius-sm: 1px;
+  --radius: 4px;
+  --radius: 4px;
```
[the remark this answers is not in the document]
Comment (shinokada) [reply static/themes/enterprise-runtime.css]: fixed.

File: static/themes/mono-runtime.css [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+  --leading-7: 28px;
+  --leading-8: 32px;
+  --leading-6: 24px;
+  --leading-4: 16px;
+  --leading-none: 1px;
+  --leading-5: 20px;
+  --tracking-tighter: -0.8px;
+  --leading-heading-none: 60px;
+  --tracking-tight: -0.4px;
+  --tracking-normal: -0.4px;
```
[the remark this answers is not in the document]
Comment (shinokada) [reply static/themes/mono-runtime.css]: fixed.

File: static/themes/playful-runtime.css [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+  --tracking-tighter: -0.8px;
+  --leading-heading-none: 60px;
+  --tracking-tight: -0.4px;
+
+  /* radius variables */
+  --radius-xxs: 2px;
+  --radius-xs: 2px;
+  --radius-sm: 10px;
+  --radius: 10px;
+  --radius: 14px;
```
[the remark this answers is not in the document]
Comment (shinokada) [reply static/themes/playful-runtime.css]: fixed.

File: static/themes/themes-runtime.css [hunk 15]
```diff
@@ -0,0 +1,1106 @@
...
+  --leading-9: 36px; /* rem pls */
+  --leading-7: 28px;
+  --leading-8: 32px;
+  --leading-6: 24px;
+  --leading-4: 16px;
+  --leading-none: 1px;
+  --leading-5: 20px;
+  --tracking-tighter: -0.8px;
+  --leading-heading-none: 60px;
+  --tracking-tight: -0.4px;
```
[the remark this answers is not in the document]
Comment (shinokada) [reply static/themes/themes-runtime.css]: fixed.

File: static/themes/themes-runtime.css [hunk 16]
```diff
@@ -0,0 +1,1106 @@
...
+    --leading-9: 36px; /* rem pls */
+    --leading-7: 28px;
+    --leading-8: 32px;
+    --leading-6: 24px;
+    --leading-4: 16px;
+    --leading-none: 1px;
+    --leading-5: 20px;
+    --tracking-tighter: -0.8px;
+    --leading-heading-none: 60px;
+    --tracking-tight: -0.4px;
```
[the remark this answers is not in the document]
Comment (shinokada) [reply static/themes/themes-runtime.css]: fixed.

File: static/themes/themes-runtime.css [hunk 17]
```diff
@@ -0,0 +1,1106 @@
...
+    --text-xs: 0.825rem;
+    --text-sm: 0.925rem;
+    --text-base: 1.125rem;
+
+    /* radius variables */
+    --radius-xxs: 1px;
+    --radius-xs: 1px;
+    --radius-sm: 1px;
+    --radius: 4px;
+    --radius: 4px;
```
[the remark this answers is not in the document]
Comment (shinokada) [reply static/themes/themes-runtime.css]: fixed.

File: static/themes/themes-runtime.css [hunk 18]
```diff
@@ -0,0 +1,1106 @@
...
+    --tracking-tighter: -0.8px;
+    --leading-heading-none: 60px;
+    --tracking-tight: -0.4px;
+
+    /* radius variables */
+    --radius-xxs: 2px;
+    --radius-xs: 2px;
+    --radius-sm: 10px;
+    --radius: 10px;
+    --radius: 14px;
```
[the remark this answers is not in the document]
Comment (shinokada) [reply static/themes/themes-runtime.css]: fixed.

File: src/tests/extend/button-toggle/button-toggle.test.ts [hunk 19]
```diff
@@ -0,0 +1,347 @@
...
+
+    test("applies size from group context", () => {
+      render(SizeGroupTest);
+      const group = screen.getByTestId("size-group");
+
+      expect(group).toBeInTheDocument();
+      expect(screen.getByTestId("size-opt1")).toBeInTheDocument();
+      expect(screen.getByTestId("size-opt2")).toBeInTheDocument();
+    });
+  });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/button-toggle/button-toggle.test.ts:424]: fixed.

File: src/lib/extend/button-toggle/CheckIcon.svelte [hunk 20]
```diff
@@ -14,6 +14,7 @@
   stroke-width="2"
   stroke-linecap="round"
   stroke-linejoin="round"
+  role="img"
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/extend/button-toggle/CheckIcon.svelte]: fixed.

File: src/lib/extend/clipboard-manager/ClipboardManager.svelte [hunk 21]
```diff
@@ -233,23 +233,25 @@
...
+      await navigator.clipboard.writeText(trimmed);
     } catch (e) {
+      // Silently handle clipboard errors (e.g., permissions, unsupported browsers)
       console.error("Clipboard write failed:", e);
-      showToast("Saved but clipboard access denied", "error");
     }
+
+    items = sortItems([item, ...items]).slice(0, limit);
+    newText = "";
+    showToast("Saved and copied to clipboard");
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/extend/clipboard-manager/ClipboardManager.svelte]: fixed.

File: src/tests/extend/clipboard-manager/clipboard-manager.test.ts [hunk 22]
```diff
@@ -0,0 +1,592 @@
...
+      await user.click(saveButton);
+
+      // Verify item was added to the UI
+      await waitFor(() => {
+        expect(screen.getByText("Copy test")).toBeInTheDocument();
+      });
+
+      // Note: clipboard.writeText mock is not reliably called in Vitest browser mode
+      // The actual clipboard functionality works in production but is difficult to test
+    });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/clipboard-manager/clipboard-manager.test.ts]: Renamed.

File: src/tests/typography/secondary/secondary.test.ts [hunk 23]
```diff
@@ -0,0 +1,140 @@
...
+      expect(secondary).toHaveClass("custom-secondary-class");
+    });
+
+    test("merges custom class with theme classes", () => {
+      const { container } = render(CustomClassSecondaryTest);
+      const secondary = container.querySelector('[data-testid="custom-secondary"]');
+
+      expect(secondary).toHaveClass("custom-secondary-class");
+      expect(secondary?.classList.length).toBeGreaterThan(0);
+    });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/typography/secondary/secondary.test.ts:56]: fixed.

File: src/tests/typography/secondary/secondary.test.ts [hunk 24]
```diff
@@ -0,0 +1,140 @@
...
+  });
+
+  describe("Semantic HTML", () => {
+    test("uses semantic small element", () => {
+      const { container } = render(BasicSecondaryTest);
+      const secondary = container.querySelector("small");
+
+      expect(secondary).toBeInTheDocument();
+      expect(secondary?.tagName).toBe("SMALL");
+    });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/typography/secondary/secondary.test.ts]: Removed.

File: src/tests/typography/secondary/secondary.test.ts [hunk 25]
```diff
@@ -0,0 +1,140 @@
...
+
+  describe("Theme Integration", () => {
+    test("applies theme classes", () => {
+      const { container } = render(BasicSecondaryTest);
+      const secondary = container.querySelector('[data-scope="secondary"]');
+
+      // Verify specific theme classes are present
+      expect(secondary?.className).toContain('text-'); // or check for specific theme class pattern
+      expect(secondary?.classList.length).toBeGreaterThan(0);
+    });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/typography/secondary/secondary.test.ts:96]: fixed.

File: src/lib/extend/clipboard-manager/theme.ts [hunk 26]
```diff
@@ -32,7 +32,7 @@ export const clipboardManager = tv({
     itemContent: "flex-1 min-w-0",
     itemHeader: "flex items-center gap-2 mb-1",
     itemTimestamp: "text-xs text-gray-500 dark:text-gray-400",
-    itemText: "text-sm text-gray-900 dark:text-gray-100 break-words line-clamp-2",
+    itemText: "text-sm text-gray-900 dark:text-gray-100 break-words line-clamp-2 text-body",
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/extend/clipboard-manager/theme.ts:35]: It is defined in css files in static/themes/ directory.

File: src/routes/docs-examples/extend/clipboard-manager/examples/WithModal.svelte [hunk 27]
```diff
@@ -8,7 +8,7 @@
 
 <ClipboardManager bind:open={clipboardModal} enableSelectionMenu={true} selectionTarget="#with-modal" storageKey="modal-clipboard" />
 
-<P id="with-modal" class="py-4">
+<P id="with-modal" class="py-4 text-white">
   Lorem ipsum dolor sit amet consectetur adipisicing elit. Aperiam, repudiandae. Optio delectus nihil assumenda laborum voluptatum nam illum nobis blanditiis esse sapiente, cumque facere ab
   consequatur. Odit, architecto enim! At!
 </P>
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/routes/docs-examples/extend/clipboard-manager/examples/WithModal.svelte:14]: Since the parent is dark, it is ok.

File: src/lib/dialog/Dialog.svelte [hunk 28]
```diff
@@ -117,25 +118,81 @@
...
+        isFocusMovedOutside = true;
+      }
+    }
 
-  const focusTrap = (node: HTMLElement) => (focustrap ? trapFocus(node) : undefined);
+    // initial focus
+    const elements = focusable();
+    if (elements.length > 0) {
+      elements[0].focus();
+    }
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/dialog/Dialog.svelte]: Removed and fixed.

File: src/tests/extend/command-palette/command-palette.test.ts [hunk 29]
```diff
@@ -0,0 +1,682 @@
...
+
+      // const searchInput = screen.getByRole("combobox") as HTMLInputElement;
+      // Assume the test component filters to empty initially or we type something
+      // For this test, let's assume EmptyStatePalette renders with search that yields no results
+
+      const emptyState = screen.queryByTestId("empty-state");
+      if (emptyState) {
+        expect(emptyState).toBeInTheDocument();
+      }
+    });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/command-palette/command-palette.test.ts]: Fixed.
The following is better without errors:
```ts
test("shows empty state with custom message", async () => {
      render(EmptyStatePalette);

      const input = screen.getByRole("combobox");
      await userEvent.type(input, "abc");

      const emptyState = screen.getByText(
        "No commands found. Try a different search."
      );

      expect(emptyState).toBeInTheDocument();
      expect(emptyState.closest("[data-part='empty-state']")).toBeTruthy();
    });
```

File: src/tests/extend/command-palette/filtering-palette.test.svelte [hunk 30]
```diff
@@ -0,0 +1,49 @@
...
+  ];
+</script>
+
+<CommandPalette bind:open {items} data-testid="filtering-palette" />
+
+{#if open}
+  <div data-testid="empty-state" data-part="empty-state">
+    <!-- This would be rendered by the component when no results -->
+  </div>
+{/if}
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/command-palette/filtering-palette.test.svelte]: Fixed.

File: src/tests/extend/kanban/card-with-tags.test.svelte [hunk 31]
```diff
@@ -0,0 +1,63 @@
...
+                <span data-testid={`card-${card.id}-tag-${i}`} data-part="tag">{tag}</span>
+              {/each}
+            </div>
+          {/if}
+        </article>
+      {/each}
+    </div>
+    <button data-testid={`add-card-btn-${column.id}`} data-part="button" aria-label={`Add card to ${column.title}`}>+ Add card</button>
+  </div>
+{/each}
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/kanban/card-with-tags.test.svelte]: Removed and fixed.

File: src/tests/extend/kanban/empty-board.test.svelte [hunk 32]
```diff
@@ -0,0 +1,34 @@
...
+
+{#each columns as column (column.id)}
+  <div data-testid={`column-${column.id}`} data-part="column" role="group" aria-label={`${column.title} column drop zone`}>
+    <h2 data-testid={`column-title-${column.id}`} data-part="title">{column.title}</h2>
+    <div data-testid={`card-list-${column.id}`} data-part="list" role="list" aria-label={`${column.title} cards`}>
+      <!-- Empty list -->
+    </div>
+    <button data-testid={`add-card-btn-${column.id}`} data-part="button" aria-label={`Add card to ${column.title}`}>+ Add card</button>
+  </div>
+{/each}
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/kanban/empty-board.test.svelte]: Removed and fixed.

File: src/tests/extend/kanban/kanban.test.ts [hunk 33]
```diff
@@ -0,0 +1,558 @@
...
+
+    test("calls onMove callback with correct parameters", () => {
+      render(CallbacksBoard);
+
+      // Simulating move through component's internal logic
+      // The test verifies the callback structure
+
+      expect(callbacksState.onMoveCalls).toEqual([]);
+    });
+  });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/kanban/kanban.test.ts:272]: fixed.

File: src/tests/extend/kanban/kanban.test.ts [hunk 34]
```diff
@@ -0,0 +1,558 @@
...
+        if (testId) ids.add(testId);
+      });
+    });
+
+    test("preserves card data after move", () => {
+      render(CallbacksBoard);
+
+      // When a card is moved, its data (title, description, tags) should remain intact
+      // This would be verified through the onMove callback receiving complete card data
+    });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/kanban/kanban.test.ts]: fixed.

File: src/tests/uiHelpers.test.ts [hunk 35]
```diff
@@ -0,0 +1,77 @@
...
+
+    const action = clickOutside(element, firstCallback);
+
+    action.update(secondCallback);
+
+    document.body.click();
+
+    expect(firstCallback).not.toHaveBeenCalled();
+    expect(secondCallback).toHaveBeenCalledTimes(1);
+  });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/uiHelpers.test.ts:76]: fixed.

File: src/tests/extend/scroll-spy/custom-href.test.svelte [hunk 36]
```diff
@@ -0,0 +1,14 @@
...
+  const items: ScrollSpyItem[] = [
+    { id: "href-section1", label: "Href Section 1", href: "#custom-href-1" },
+    { id: "href-section2", label: "Href Section 2", href: "#custom-href-2" }
+  ];
+</script>
+
+<ScrollSpy {items} data-testid="custom-href-scrollspy" />
+
+<div id="href-section1" data-testid="href-section-1">Href Section 1 Content</div>
+<div id="href-section2" data-testid="href-section-2">Href Section 2 Content</div>
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/scroll-spy/custom-href.test.svelte]: fixed.

File: src/tests/extend/split-pane/split-pane.test.ts [hunk 37]
```diff
@@ -0,0 +1,438 @@
...
+    test("receives size array in callback", () => {
+      render(CallbackTest);
+
+      // Wait for initialization
+      if (callbackTestState.lastSizes.length > 0) {
+        expect(callbackTestState.lastSizes).toHaveLength(2);
+        expect(callbackTestState.lastSizes.every((size) => typeof size === "number")).toBe(true);
+      }
+    });
+  });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/split-pane/split-pane.test.ts:163]: Fixed.

File: src/tests/extend/scroll-spy/scroll-spy.test.ts [hunk 38]
```diff
@@ -0,0 +1,450 @@
...
+    });
+  });
+
+  describe("Callback Functions", () => {
+    test("calls onActiveChange when active section changes", () => {
+      render(CallbackTest);
+
+      // Initially should set first item as active
+      expect(callbackTestState.activeChangeCallCount).toBeGreaterThanOrEqual(0);
+    });
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/scroll-spy/scroll-spy.test.ts:206]: Suggested improvement is not correct. 
Todo for me.

File: src/lib/dialog/Dialog.svelte [hunk 39]
```diff
@@ -117,25 +118,75 @@
...
-        dlg.focus(); // fallback
-      }
+      autofocusEl ? autofocusEl.focus() : dlg.focus();
     });
 
     return () => dlg.close();
-  }
+  };
+
+  // const focusTrap = (node: HTMLElement) => (focustrap ? trapFocus(node) : undefined);
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/dialog/Dialog.svelte]: fixed.

File: src/tests/uiHelpers.test.ts [hunk 40]
```diff
@@ -0,0 +1,77 @@
...
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/uiHelpers.test.ts:76]: fixed.

File: src/lib/dialog/Dialog.svelte [hunk 41]
```diff
@@ -117,25 +117,78 @@
...
+          first?.focus();
+          event.preventDefault();
+        }
+      }
+
+      if (event.key === "Escape") {
+        event.preventDefault();
+        skipFocusRestore = true;
+        close();
+      }
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/lib/dialog/Dialog.svelte:169]: Removed.

File: svelte.config.js [hunk 42]
```diff
@@ -11,7 +11,9 @@ import { includeFiles } from "./include-files.js";
 const config = {
   extensions: [".svelte", ...mdsvexConfig.extensions],
   compilerOptions: {
-    // runes: true
+    experimental: {
+      async: true
+    }
```
[the remark this answers is not in the document]
Comment (shinokada) [reply svelte.config.js:17]: fixed.

File: vitest-setup-client.ts [hunk 43]
```diff
@@ -2,6 +2,14 @@
...
 import "@testing-library/jest-dom/vitest";
 import { beforeEach, afterEach, vi } from "vitest";
+import { installPopoverPolyfill } from "./src/tests/utils/installPopoverPolyfill";
+import { prefersReducedMotion } from "svelte/motion";
+
+Object.defineProperty(prefersReducedMotion, "current", {
+  get: () => true
+});
+
+installPopoverPolyfill();
```
[the remark this answers is not in the document]
Comment (shinokada) [reply vitest-setup-client.ts]: Updated.
```ts
import "@testing-library/jest-dom/vitest";
import { beforeEach, afterEach, vi } from "vitest";

const mockMatchMedia = (initialMatches = false): MediaQueryList & {
  dispatchChange: (matches: boolean) => void;
} => {
  let _matches = initialMatches;

  const listeners = new Set<(event: MediaQueryListEvent) => void>();

  const mql: MediaQueryList & {
    dispatchChange: (matches: boolean) => void;
  } = {
    media: "(prefers-reduced-motion: reduce)",
    onchange: null,

    get matches() {
      return _matches;
    },

    addEventListener(
      _type: string,
      listener: EventListenerOrEventListenerObject
    ) {
      if (typeof listener === "function") {
        listeners.add(listener as (e: MediaQueryListEvent) => void);
      } else {
        listeners.add((e) => listener.handleEvent(e));
      }
    },

    removeEventListener(
      _type: string,
      listener: EventListenerOrEventListenerObject
    ) {
      if (typeof listener === "function") {
        listeners.delete(listener as (e: MediaQueryListEvent) => void);
      } else {
        listeners.delete((e) => listener.handleEvent(e));
      }
    },

    // Legacy APIs (required by TS)
    addListener(
      listener: (this: MediaQueryList, ev: MediaQueryListEvent) => any
    ) {
      listeners.add(listener);
    },

    removeListener(
      listener: (this: MediaQueryList, ev: MediaQueryListEvent) => any
    ) {
      listeners.delete(listener);
    },

    dispatchEvent(event: Event) {
      if (!("matches" in event)) return false;
      listeners.forEach((listener) =>
        listener.call(this, event as MediaQueryListEvent)
      );
      return true;
    },

    dispatchChange(newMatches: boolean) {
      _matches = newMatches;

      const event = { matches: newMatches } as MediaQueryListEvent;

      listeners.forEach((listener) => listener.call(this, event));
      this.onchange?.call(this, event);
    },
  };

  return mql;
};

const mm = mockMatchMedia(true);

Object.defineProperty(window, "matchMedia", {
  configurable: true,
  writable: true,
  value: (query: string) => {
    if (query.includes("prefers-reduced-motion")) {
      return mm;
    }

    // Default behavior for everything else
    return mockMatchMedia(false);
  },
});


// import here
import { installPopoverPolyfill } from "./src/tests/utils/installPopoverPolyfill";
installPopoverPolyfill();
...
```

File: src/tests/mega-menu/mega-menu.test.ts [hunk 44]
```diff
@@ -49,36 +49,23 @@ describe("MegaMenu - Full Width", () => {
...
 
   test("renders extra content in full width mode", async () => {
     const user = userEvent.setup();
     render(FullMegaMenu);
 
-    // Wait for component to mount
-    await new Promise((resolve) => setTimeout(resolve, 50));
-
-    const trigger = screen.getByTestId("full-mega-menu-trigger");
+    const trigger = await screen.getByTestId("full-mega-menu-trigger");
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/mega-menu/mega-menu.test.ts]: Fixed.

File: src/tests/extend/command-palette/command-palette.test.ts [hunk 45]
```diff
@@ -0,0 +1,684 @@
...
+
+      // Wait for the dialog to be removed
+      await waitFor(
+        () => {
+          expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
+        },
+        { timeout: 3000 }
+      );
+    });
+
```
[the remark this answers is not in the document]
Comment (shinokada) [reply src/tests/extend/command-palette/command-palette.test.ts:113]: Both updates produce errors.

File: reactive-context.md [hunk 46]
```diff
@@ -0,0 +1,60 @@
+# Reactive Context
+
+## Pattern 1: Object with Getters (Accordion)
+Better for: Multiple reactive properties that need to be accessed independently
+```ts
+typescriptconst reactiveCtx = {
```
[the remark this answers is not in the document]
Comment (shinokada) [reply reactive-context.md]: fixed.

File: reactive-context.md [hunk 47]
```diff
@@ -0,0 +1,60 @@
...
+The downside is this creates a new object on each access, which could cause unnecessary re-renders if used in places expecting referential equality.
+
+## Conclusion
+Use Pattern 1 for multi-property contexts, Pattern 2 for single values. Both are "correct" - choose based on ergonomics for consistency vs. optimization.
+
+## Reference
+
+- https://svelte.dev/docs/svelte/$effect#When-not-to-use-$effect
+- https://svelte.dev/docs/svelte/compiler-warnings#state_referenced_locally
+- https://svelte.dev/docs/svelte/await-expressions
```
[the remark this answers is not in the document]
Comment (shinokada) [reply reactive-context.md]: fixed
